### PR TITLE
Removing TODOs from Refactored LLCP 

### PR DESCRIPTION
**Pull request by erbr-ot**
This removes a bunch of TODOs ranging in complexity from no-brainers to minor modifications.

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/ull_llcp_remote.c [hunk 1]
```diff
@@ -73,37 +73,24 @@ enum {
...
 	case PROC_LE_PING:
-		return 0U;
 	case PROC_VERSION_EXCHANGE:
-		return 0U;
 	case PROC_ENCRYPTION_START:
 	case PROC_ENCRYPTION_PAUSE:
+	case PROC_TERMINATE:
+	case PROC_DATA_LENGTH_UPDATE:
+	case PROC_CTE_REQ:
+		return 0U;
```
Comment (kruithofa) [inline subsys/bluetooth/controller/ll_sw/ull_llcp_remote.c:87]: Please remove one return statement

**Review by kruithofa — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/ll_sw/ull_llcp_remote.c (hunk 1 above)
Comment (erbr-ot) [reply subsys/bluetooth/controller/ll_sw/ull_llcp_remote.c:87]: Yeap

**Review by kruithofa — DISMISSED** [review]

**Review by sjanc — DISMISSED** [review]

**Review by carlescufi — DISMISSED** [review]

File: subsys/bluetooth/controller/ll_sw/ull_llcp_conn_upd.c [hunk 2]
```diff
@@ -44,10 +44,6 @@
 /* Hardcoded instant delta +6 */
 #define CONN_UPDATE_INSTANT_DELTA 6U
 
-/* TODO: Known, missing items (missing implementation):
```
Comment (erbr-ot) [inline subsys/bluetooth/controller/ll_sw/ull_llcp_conn_upd.c:47]: Do not remove

File: subsys/bluetooth/controller/ll_sw/ull_llcp_common.c [hunk 3]
```diff
@@ -933,6 +933,10 @@ static void rp_comm_send_rsp(struct ll_conn *conn, struct proc_ctx *ctx, uint8_t
 	case PROC_VERSION_EXCHANGE:
 		/* The Link Layer shall only queue for transmission a maximum of one
 		 * LL_VERSION_IND PDU during a connection.
+		 * If the Link Layer receives an LL_VERSION_IND PDU and has already sent an
+		 * LL_VERSION_IND PDU then the Link Layer shall not send another
+		 * LL_VERSION_IND PDU to the peer device.
+		 * So - simply ignore if already sent
```
Comment (erbr-ot) [inline subsys/bluetooth/controller/ll_sw/ull_llcp_common.c:939]: Include term: "invalid behaviour"

File: tests/bluetooth/controller/mock_ctrl/src/util.c [hunk 4]
```diff
@@ -18,8 +18,8 @@
 /**
  * @brief Population count: Count the number of bits set to 1
  * @details
- * TODO: Faster methods available at [1].
- * [1] http://graphics.stanford.edu/~seander/bithacks.html#CountBitsSetParallel
+ * FYI - Faster methods available at
```
Comment (erbr-ot) [inline tests/bluetooth/controller/mock_ctrl/src/util.c:21]: remove

File: tests/bluetooth/controller/ctrl_hci/src/main.c [hunk 5]
```diff
@@ -131,13 +137,12 @@ void test_hci_feature_exchange_wrong_handle(void)
 
 	zassert_equal(err, BT_HCI_ERR_UNKNOWN_CONN_ID, "Wrong reply for wrong handle\n");
 
+	/* Use up all local procedure contexts */
 	ctx_counter = 0;
 	do {
 		ctx = llcp_create_local_procedure(PROC_FEATURE_EXCHANGE);
 		ctx_counter++;
 	} while (ctx != NULL);
-	zassert_equal(ctx_counter, test_ctx_buffers_cnt() + 1,
```
Comment (erbr-ot) [inline tests/bluetooth/controller/ctrl_hci/src/main.c:139]: remove?

Comment (erbr-ot) [reply tests/bluetooth/controller/ctrl_hci/src/main.c:139]: Removal of this check is ok, as the point of the test is to confirm that after 'using all resources' an attempt to initiate the procedure will fail with CMD_DISALLOWED. And after splitting resource into local/remote there is no published API for getting the number of local procedure ctx allocated, so this check is simply removed

**Review by kruithofa — APPROVED** [review]

**Review by thoh-ot — APPROVED** [review]

**Pull request merged**